Websource –  https://www.kaggle.com/datasets/protobioengineering/united-states-fda-drugs-feb-2024/data
Date accessed: 10th October 2024
All FDA Drugs (United States, 1939-Present)
credit to Proto Bioengineering on kaggle.com


In [16]:
# install the required libraries for data analysis
import numpy as np
import pandas as pd
import datetime as dt

# read the data from the drugs.csv files obtained from the kaggle website

1. Read data from the drugs.csv file

In [17]:
# read data from the drugs.csv file and save it in a variable called drugs_data
drugs_data = pd.read_csv('/Users/h.nderitu/PyData/Capstone Project/Pydata Capstone Downloaded Datasets/All FDA Drugs (United States, 1939-Present)/drugs.csv')
print(drugs_data)


      application_number         sponsor_name                  brand_name  \
0              NDA000159                LILLY               SULFAPYRIDINE   
1             ANDA076191                 BARR  FEXOFENADINE HYDROCHLORIDE   
2             ANDA076191                 BARR  FEXOFENADINE HYDROCHLORIDE   
3             ANDA076191                 BARR  FEXOFENADINE HYDROCHLORIDE   
4             ANDA076206       ROCKWELL MEDCL                  CALCITRIOL   
...                  ...                  ...                         ...   
48211         ANDA076134   SUN PHARM INDS LTD                  LORATADINE   
48212         ANDA076139                 TEVA           FOSINOPRIL SODIUM   
48213         ANDA076139                 TEVA           FOSINOPRIL SODIUM   
48214         ANDA076139                 TEVA           FOSINOPRIL SODIUM   
48215         ANDA076150  INTL MEDICATION SYS                   LORAZEPAM   

       product_number reference_drug                  active_ingredients  \

In [19]:
# check the number of rows, entries and datatypes for the drugs_data dataframe
drugs_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48216 entries, 0 to 48215
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   application_number   48216 non-null  object
 1   sponsor_name         48216 non-null  object
 2   brand_name           48216 non-null  object
 3   product_number       48216 non-null  int64 
 4   reference_drug       48216 non-null  object
 5   active_ingredients   48216 non-null  object
 6   referenced_standard  46889 non-null  object
 7   dosage_form          48215 non-null  object
 8   route                48130 non-null  object
 9   marketing_status     48216 non-null  object
dtypes: int64(1), object(9)
memory usage: 3.7+ MB


10 columns

reference standard column has 46889 entries - 1325 missing values

dosage form column has 48215 entries - 1 missing value

route column has 48130 enries - 86 missing values




# check for missingness. 

In [33]:
#check which columns are missing values
drugs_data.isnull().any()

application_number     False
sponsor_name           False
brand_name             False
product_number         False
reference_drug         False
active_ingredients     False
referenced_standard     True
dosage_form             True
route                   True
marketing_status       False
dtype: bool

3 columns missing values: referenced_ standard, dosage_form and route. 

In [30]:
# check number of rows with missing values. 
drugs_data.isnull().any(axis=1).value_counts()

False    46847
True      1369
Name: count, dtype: int64

In [31]:
# check number of columns with missing values. 
drugs_data.isnull().any(axis=0).value_counts()

False    7
True     3
Name: count, dtype: int64

In [38]:
# check and select all rows and columns with missing values. 
missing_col_drugs = drugs_data[drugs_data.isnull().any(axis=1)]
print(missing_col_drugs)

      application_number          sponsor_name              brand_name  \
77            ANDA076558                 HIKMA           CIPROFLOXACIN   
84            ANDA076596           TEVA PHARMS               MODAFINIL   
85            ANDA076596           TEVA PHARMS               MODAFINIL   
213           ANDA077121    NORVIUM BIOSCIENCE                BACLOFEN   
292           ANDA077412           TEVA PHARMS    ESCITALOPRAM OXALATE   
...                  ...                   ...                     ...   
47755         ANDA071233               VINTAGE             HALOPERIDOL   
47803         ANDA071994  IVAX SUB TEVA PHARMS  PRAZOSIN HYDROCHLORIDE   
47891         ANDA074047                  BARR              ZIDOVUDINE   
48026         ANDA075040           ESI LEDERLE                PROPOFOL   
48032         ANDA075075    MYLAN TECHNOLOGIES           NITROGLYCERIN   

       product_number reference_drug                    active_ingredients  \
77                  1            

In [20]:
drugs_data.head()

application_number    sponsor_name                  brand_name  \
0          NDA000159           LILLY               SULFAPYRIDINE   
1         ANDA076191            BARR  FEXOFENADINE HYDROCHLORIDE   
2         ANDA076191            BARR  FEXOFENADINE HYDROCHLORIDE   
3         ANDA076191            BARR  FEXOFENADINE HYDROCHLORIDE   
4         ANDA076206  ROCKWELL MEDCL                  CALCITRIOL   

   product_number reference_drug                  active_ingredients  \
0               1             No               SULFAPYRIDINE (500MG)   
1               2             No   FEXOFENADINE HYDROCHLORIDE (60MG)   
2               3             No  FEXOFENADINE HYDROCHLORIDE (180MG)   
3               1             No   FEXOFENADINE HYDROCHLORIDE (30MG)   
4               1             No             CALCITRIOL (0.001MG/ML)   

  referenced_standard dosage_form      route marketing_status  
0                  No      TABLET       ORAL     Discontinued  
1                  No      TABLET       ORAL     Discontinued  
2                  No      TABLET       ORAL     Discontinued  
3                  No      TABLET       ORAL     Discontinued  
4                  No  INJECTABLE  INJECTION     Discontinued